openquake/sep/liquefaction/liquefaction.py

```diff
@@ -113,18 +152,496 @@ def zhu_liquefaction_probability_general(
+    wtd: Union[float, np.ndarray],
+    precip: Union[float, np.ndarray],
+    intercept: float = 8.801,
+    pgv_coeff: float = 0.334,
+    vs30_coeff: float = -1.918,
+    dw_coeff: float = -0.2054,
+    wtd_coeff: float = -0.0333,
+    precip_coeff: float = 0.0005408
+) -> Union[float, np.ndarray]:
+    """
+    Calculates the probability of a site undergoing liquefaction using the
+    logistic regression of the Zhu et al., 2017 noncoastal model, which is the 
+    model recommended by the authors for global implementation, as modified 
+    by Rashidian and Baise (2020) to decrease over-prediction in large areas 
+    experiencing low PGA (below 0.1 g) with the addition of a PGA threshold 
+    (no liquefaction when PGA < 0.1 g). For use in their the US Geological 
+    Survey (USGS) ground failure (GF) product, further caps on precipitation 
+    and PGV of 2500 mm/year and 150 cm/s respectively were imposed to avoid
+    unrealistic extrapolations when either factor was much higher than found 
+    in the training data, as described in Allstadt et al. (2022). Finally, an
+    ad-hoc magnitude scaling factor is applied to the PGV values to reduce
+    overprediction of liquefaction probabilities for lower magnitude events.
+    The optimal threshold probability value to convert the predicted probability
+    into binary classification is 0.4 (see p.13 from Zhu et al., 2017).
+
+    Reference: Allstadt, K. E., Thompson, E. M., Jibson, R. W., Wald, D. J., 
+    Hearne, M., Hunter, E. J., Fee, J., Schovanec, H., Slosky, D., 
+    & Haynie, K. L. (2022). The US Geological Survey ground failure product: 
+    Near-real-time estimates of earthquake-triggered landslides and liquefaction. 
+    Earthquake Spectra, 38(1), 5–36. https://doi.org/10.1177/87552930211032685
+
+    :param pgv:
+        Peak Ground Velocity, measured in cm/s
+    :param vs30:
+        Shear-wave velocity averaged over the upper 30 m of the earth at the
+        site, measured in m/s
+    :param dw:
+        Distance to the nearest water body, measured in km
+    :param wtd:
+        Global water table depth, measured in m
+    :param precip:
+        Mean annual precipitation, measured in mm
+
+    :returns:
+        prob_liq: Probability of liquefaction at the site.
+        out_class: Binary output 0 or 1, i.e., liquefaction nonoccurrence
+                   or liquefaction occurrence occurrence.
+    """
+    pgv = np.where(pgv > 150, 150, pgv)
+    precip = np.where(precip > 2500, 2500, precip)
+    pgv_scaling_factor = 1.0 / (1.0 + np.exp(-2.0 * (mag - 6.0)))
+    Xg = (pgv_coeff * np.log(pgv_scaling_factor * pgv) 
+          + vs30_coeff * np.log(vs30) + precip_coeff * precip 
+          + dw_coeff * dw + wtd_coeff * wtd + intercept)
+    prob_liq = sigmoid(Xg)
+
+    # prob_liq = rashidian_baise_2020_liquefaction_probability(pga, pgv, vs30, dw, wtd, precip, 
+    #              intercept=intercept, pgv_scaling_factor=pgv_scaling_factor, 
```

Comment:
Do we need the commented lines or are they a leftover of a previous implementation?

Reply:
Solved.